tests/webTables.spec.ts

```diff
@@ -361,72 +224,49 @@ test('Validate specialty lists', async ({ page }) => {
     // Navigate to the Specialties page
     await page.getByRole('link', {name: 'Specialties'}).click()
 
-    // Wait for all table contents to load
-    await page.waitForTimeout(timeout)
-
+    await page.waitForSelector('tbody')
     /* 2. On the Specialties page, select "Add" button. Type the new specialty "oncology" and click "Save" button */
     // Click on 'Add' button to enter a new specialty, oncology
     await page.getByRole('button', {name: 'Add'}).click()
     await page.locator('#name').fill('oncology')
     await page.getByText('Save').click()
 
-    await page.waitForTimeout(timeout)
+    // Wait for Specialty table contents to load
+    await page.waitForSelector('tbody')
+    
+    // Validate 'oncology' was added
+    await expect(page.locator('tbody tr').last().getByRole('textbox')).toHaveValue('oncology')
 
-    /* 3. Extract all values of specialties and put them into the array. */
+     /* 3. Extract all values of specialties and put them into the array. */
     // Collect all specialties and store for later usage
-    const allSpecialtyRows = await page.getByRole('row').all()
-    let allSpecialties: string[] = []
-
-    // Iterate through each table row to store specialties
-    for(const row of allSpecialtyRows){
-        const rowData = await row.getByRole('textbox').all()
+    const allSpecialtyInputs = await page.locator('tbody tr').getByRole('textbox').all()
+    const allSpecialties : string [] = []
 
-        // Skip the header
-        if(rowData.length > 0){
-            // Add first cell of row data (the specialty name) to the array
-            allSpecialties.push(await rowData[0].inputValue())
-        }
+    // Iterate through all the rows to collect the Specialty and store in allSpecialties for future validation
+    for(const input of allSpecialtyInputs){
+        console.log(await input.inputValue())
+        allSpecialties.push(await input.inputValue())
     }
 
     /* 4. Select the VETERINARIANS menu item in the navigation bar, then select "All" */
     // Navigate to the 'Veterinarians' page through the dropdown menu at the top
     await page.getByText("Veterinarians").click();
     await page.getByText("All").click()
 
-    // Wait for the Vets page to load
-    await page.waitForTimeout(timeout)
+    // Wait for all table contents to load
+    await page.waitForSelector('table')
```

Comment:
This wait is not needed either. The next command is a `click()`, it has auto-waiting feature as well